packages/react-auth/src/ComponentAuthProvider/index.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  FC,
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useState,
+  useMemo,
+  useEffect,
+} from "react";
+
+import AuthNClient from "@src/AuthNClient";
+
+import { APIResponse, AuthNConfig, Token, AuthUser } from "@src/types";
+
+export interface SessionData {
+  user?: AuthUser;
+}
+
+export interface ComponentAuthContextType {
+  authenticated: boolean;
+  loading: boolean;
+  error: APIResponse | undefined;
+  user?: AuthUser;
+  login: () => void;
+  logout: () => void;
+  config: AuthNConfig;
```

Comment:
should both provider just make the memoized client object available to the users.

I assume the client will grow to include all client endpoints like the session endpoints, and users may want to be all to easily call something like session list if they write a component to show users everywhere they are logged in